frame/node-authorization/src/lib.rs

```diff
@@ -90,21 +94,70 @@ pub mod pallet {
 		type WeightInfo: WeightInfo;
 	}
 
+	/// A bounded opaque peer ID.
+	#[derive(Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	#[codec(mel_bound())]
+	#[scale_info(skip_type_params(T))]
+	pub struct BoundedPeerId<T: Config>(BoundedVec<u8, T::MaxPeerIdLength>);
```

Comment:
I think we should really try to give an upper bound on `PeerId` itself, rather than creating a container type around it to sidestep the issue. @shawntabrizi How exactly should we make primitive types bounded?